app/bootstrap/internal/k8s/k8s.go

```diff
@@ -0,0 +1,193 @@
+//    \\ SPIKE: Secure your secrets with SPIFFE. — https://spike.ist/
+//  \\\\\ Copyright 2024-present SPIKE contributors.
+// \\\\\\\ SPDX-License-Identifier: Apache-2.0
+
+// Package k8s provides utilities for managing bootstrap state in Kubernetes
+// environments. It handles coordination between multiple bootstrap instances
+// to ensure bootstrap operations run exactly once per cluster.
+package k8s
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spiffe/spike-sdk-go/config/env"
+	"github.com/spiffe/spike-sdk-go/log"
+	k8s "k8s.io/api/core/v1"
+	k8sMeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const k8sTrue = "true"
```

Comment:
set `"true"` as a constant because we want "exact" case-sensitive match.

That's what I observe in some of the k8s controllers etc: Exact case-sensitive match for booleans.